engine/src/main/resources/assets/schemas/schemaClip.json

```diff
@@ -0,0 +1,68 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Clips are the ITEMS loaded into guns. They specify things like buying price, display name...",
+  "type": "object",
+  "required": [
+    "projectile",
+    "size"
+  ],
+  "properties": {
+    "projectile": {
+      "description": "name of the projectile shot by this clip, defined in projectilesConfig",
+      "type": "string",
+      "pattern": "^\\w+$"
+    },
+    "infinite": {
+      "type": "boolean",
+      "description": "True if it never runs out, false otherwise. When false, several new fields are required"
+    },
+    "size": {
+      "type": "integer",
+      "exclusiveMinimum": 0,
+      "description": "Amount of projectiles in the clip"
+    },
+    "projectilesPerShot": {
+      "type": "integer",
+      "minimum": 1,
+      "description": "Amount of projectiles consumed per shot"
+    }
+  },
+  "anyOf": [
+    {
+      "properties": {
+        "infinite": {
+          "enum": [
+            true
+          ]
+        }
+      },
+      "required": ["infinite"]
+    },
+    {
+      "properties": {
+        "infinite": {
+          "enum": [
+            false
+          ]
+        },
+        "price": {
```

Comment:
Should the price have a minimum? I tried putting in a negative price and the game still ran, and I made money by firing bullets then buying more 😄 . Negative price _technically_ works, but I don't think it should be allowed...

Reply:
Also the existence of a price minimum is inconsistant between all schemas, so what is decided should probably be applied to all price fields.

Reply:
If you want a module to add enemy-healing bullets that give you money for buying them, why not? :wink: Consistence should however be seeked in the future.

Reply:
If negative price should be allowed, can you remove the exlcusive minimums from the other schemas in this PR? I believe they are only in `schemaGun.json` and `schemaAbilityCharges.json`, shouldn't be too much of an issue 😄